app/src/main/java/com/chouten/app/ui/views/info/InfoPageViewModel.kt

```diff
@@ -69,96 +69,127 @@ class InfoPageViewModel(
     val infoResults: List<InfoResult.MediaListItem>
         get() = infoResult
 
-    init {
-        // Both title and url are url-encoded.
-        title = URLDecoder.decode(title, "UTF-8")
-        val decodedUrl = URLDecoder.decode(url, "UTF-8")
+    fun getCode(): String{
```

Comment:
Please refer to the comment regarding the Search viewmodel. The same thing can be done here. If the same code is required multiple times, consider moving it within the module layer itself.